packages/node/src/index.ts

```diff
@@ -416,7 +416,29 @@ export const build: BuildV3 = async ({
   let isEdgeFunction = isMiddleware;
 
   const project = new Project();
-  const staticConfig = getConfig(project, entrypointPath);
+  const objectConfig = getConfig(project, entrypointPath);
+  const segmentConfig = getSegmentConfig(project, entrypointPath);
+
+  if (objectConfig && segmentConfig) {
+    // If a user tries to customize in both handler and vercel.json we throw
+    const segmentConfigKeys = new Set(Object.keys(segmentConfig));
+    const duplicateKeys = Object.keys(objectConfig).filter(key =>
+      segmentConfigKeys.has(key)
+    );
+
+    if (duplicateKeys.length > 0) {
```

Comment:
I'm not sure this is the right solution.

Doesn't `vercel.json` let you use a glob pattern to match many files?

Reply:
Correct me if I'm wrong but I don't think `maxDuration` in `vercel.json` is respected in our Node runtime at all currently. The `vercel.json` config should be passed to the `build` function as `build` prop, but in the current implementation (https://github.com/vercel/vercel/blob/main/packages/node/src/index.ts) I don't see where this value would be consumed. In regards to `getConfig` and `getSegmentConfig` — they do static code analysis on the handler file directly. This line of code basically prevents doing something like this:

```
export const config = { memory: 128, maxDuration: 1 };
export const maxDuration = 5;
```

Reply:
It has been supported in `vercel.json` for 4 years

https://vercel.com/blog/customizing-serverless-functions